MetaMask/metamask-extension
# Optimism Bedrock Fix + New Gas Displays

## Explanation

### This PR has lots of changes, but most of them are i18n messages and fixes to existing e2e and unit tests. So it's not all that daunting. Thanks for the review ;) 

**Problem**: the Optimism Bedrock upgrade, which adds support to EIP1559, happened on June 6th.
Now, we see some issues with the current gas displays

- whenever we send a transaction Optimism network we often display  `0` on any the suggested gas fees options. This seems to be a rounding issue - fees are so low in the L2 that we end up displaying a `0` for all the options

The bug in code is that we always try to display the values in ETH, But the Optimism fees are sent from the backend in GWEI, the fix is to check if the ETH conversion is zero and then use GWEI conversion, if that is also zero, we then use WEI conversion.

- See: #19561 for more information and design considerations.
- Fixes: #19561

## Screenshots/Screencaps

### Before

![image](https://github.com/MetaMask/metamask-extension/assets/54408225/b5587345-147d-4112-b501-0793e1ead6ad)

### After

<img width="339" alt="Screenshot 2023-07-11 at 12 37 55" src="https://github.com/MetaMask/metamask-extension/assets/44811/b2cd05f6-9623-4d0e-afb2-1d5e901f870b">


<img width="362" alt="Screenshot 2023-07-11 at 12 38 13" src="https://github.com/MetaMask/metamask-extension/assets/44811/f62dd3a1-054c-4ae9-b730-8fdbe1b8c2f1">


<img width="358" alt="Screenshot 2023-07-11 at 12 38 05" src="https://github.com/MetaMask/metamask-extension/assets/44811/0f432678-54ce-483d-81c0-76114535f256">


## Manual Testing Steps

1. Select Optimism network
2. Send EIP1559
3. Click Edit fees
4. See recommended fees are 0

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
